### PR TITLE
App and App's network stability improvement.

### PR DESCRIPTION
**Pull request by adarsh-zededa**
Issue: DA container were often noticed to be in an error state with `one of the application's tasks has crashed - please restart application instance` message. 

Reasons:
socket.gaierror: [Errno -3] Try again: The name server returned a temporary failure indication

Fix:
Increased timeout value and added retries in app's `resolv.conf`

Also, fixed the case where EVE was appending the same dns config in `/mnt/rootfs/etc/resolv.conf` every time the AppInstance rebooted. 

cc: @gkodali-zededa 
Signed-off-by: adarsh-zededa <adarsh@zededa.com>

### CODE REVIEW

File: pkg/pillar/hypervisor/xen.go [hunk 1]
```diff
@@ -456,7 +456,9 @@ func (ctx xenContext) Info(domainName string, domainID int) (int, types.SwState,
 
 	status, err := ioutil.ReadFile("/run/tasks/" + domainName)
 	if err != nil {
-		log.Errorf("couldn't read task status file: %v", err)
+		err = fmt.Errorf("Info(%s, %d): couldn't read task status file: %v", domainName, domainID, err)
```
Comment (rvs) [inline pkg/pillar/hypervisor/xen.go:459]: I don't understand this part -- the file is guaranteed to be there simply based on the sequencing of events in domaingr and xen-start script. Please get to the bottom of what really is going on -- I don't think we can merge this workaround.

File: pkg/pillar/cmd/zedrouter/dnsmasq.go [hunk 2]
```diff
@@ -273,9 +273,9 @@ func createDnsmasqConfiglet(
 		dhcpRange = netconf.DhcpRange.Start.String()
 	}
 	if isIPv6 {
-		file.WriteString(fmt.Sprintf("dhcp-range=::,static,0,60m\n"))
+		file.WriteString(fmt.Sprintf("dhcp-range=::,static,0,12h\n"))
```
Comment (rvs) [inline pkg/pillar/cmd/zedrouter/dnsmasq.go:276]: Hey @eriknordmark @gkodali-zededa -- what's the downside of bumping it to 12h?

File: pkg/pillar/types/types.go [hunk 3]
```diff
@@ -141,7 +141,7 @@ func (state SwState) ZSwState() info.ZSwState {
 	// we map BROKEN to HALTING to indicate that EVE has an active
 	// role in reaping BROKEN domains and transitioning them to
 	// a final HALTED state
-	case BROKEN:
+	case BROKEN, UNKNOWN:
```
Comment (rvs) [inline pkg/pillar/types/types.go:144]: This is semantically incorrect

File: pkg/xen-tools/xen-start [hunk 4]
```diff
@@ -46,7 +46,10 @@ echo 0 > /dev/unknownStateCounter
 (while true; do
    sleep 11
    /etc/xen/scripts/xen-info "$1" > "/run/tasks/$1.tmp"
-   mv "/run/tasks/$1.tmp" "/run/tasks/$1"
+   if [ -s "/run/tasks/$1.tmp" ]
```
Comment (rvs) [inline pkg/xen-tools/xen-start:49]: I don't understand this logic -- the right thing to do is to fix whatever may be broken in xen-info

File: pkg/xen-tools/Dockerfile.in [hunk 5]
```diff
@@ -10,6 +10,8 @@ COPY initrd/init-initrd initrd/mount_disk.sh initrd/udhcpc_script.sh /
 COPY initrd/poweroff /sbin/poweroff
 COPY initrd/chroot2.c /tmp/
 COPY initrd/00000080 /etc/acpi/PWRF/
+COPY initrd/ssh_config /etc/ssh/
```
Comment (rvs) [inline pkg/xen-tools/Dockerfile.in:13]: Why are you adding ssh to the initrd? (well I'm pretty sure I know the answer which is debugging -- but that doesn't belong in this PR -- that belongs in your private repo).

**Review by rvs — CHANGES_REQUESTED** [review]
Hey @adarsh-zededa -- aside from dnsmasq changes -- the rest feels like a bunch of experiments that I don't think need to be in the PR. Please address my comments.

File: pkg/pillar/cmd/zedrouter/dnsmasq.go (hunk 2 above)
Comment (gkodali-zededa) [reply pkg/pillar/cmd/zedrouter/dnsmasq.go:276]: Since, we changed udhcpc_script.sh to not touch interface IP/mask unless there is change, this change may not be required. IP/mask that we make dnsmasq to offer is static, and the app/container should always receive the same IP/mask with each DHCP renewal.
One side effect of having longer lease time will be the zombie leases on devices in test setups, where we keep adding/removing apps at high frequency. At the same time, we have also patched dnsmasq to force clear leases when we decide to re-use IP addresses that are part of existing leases. So, the effect should be minimal in this case.

File: pkg/pillar/hypervisor/xen.go (hunk 1 above)
Comment (adarsh-zededa) [reply pkg/pillar/hypervisor/xen.go]: Removed.

File: pkg/pillar/types/types.go (hunk 3 above)
Comment (adarsh-zededa) [reply pkg/pillar/types/types.go]: Removed.

File: pkg/xen-tools/xen-start (hunk 4 above)
Comment (adarsh-zededa) [reply pkg/xen-tools/xen-start]: Removed. 
Will fix this in a separate PR after investigating the cause.

File: pkg/xen-tools/Dockerfile.in (hunk 5 above)
Comment (adarsh-zededa) [reply pkg/xen-tools/Dockerfile.in]: Removed.

File: pkg/pillar/cmd/zedrouter/dnsmasq.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedrouter/dnsmasq.go:276]: @adarsh-zededa can you check if the issue is gone with @gkodali-zededa 's fix to the script?

**Comment by eriknordmark** [comment]
Can you check if we need the lease time change with Gopi's PR in place?

**Comment by adarsh-zededa** [comment]
> Can you check if we need the lease time change with Gopi's PR in place?

@nordmark Reverted lease time changes as apps were stable for >5hrs without my changes.

**Review by eriknordmark — APPROVED** [review]
LGTM. The things @rvs had issues with have been removed so this just improves the reliability of the DNS setup in the domU.
